astropy/time/formats.py

```diff
@@ -942,6 +942,31 @@ class TimeYearDayTime(TimeISO):
                 '{year:d}:{yday:03d}'))
 
 
+class TimeDatetime64(TimeISOT):
+    name = 'datetime64'
+    def _check_val_type(self, val1, val2):
```

Comment:
This should just check the `dtype` of the `val1` itself, not every element.  For `datetime64` this is a built-in numpy dtype so every element is sure to have the same dtype.  This is different from `datetime`, where `val1` will be an array of objects (dtype=`'O'`).